specification/workloads/resource-manager/Microsoft.Workloads/connectors/preview/2023-10-01-preview/backupTypes.json

```diff
@@ -0,0 +1,849 @@
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/TieringPolicy"
+          }
+        },
+        "instantRpRetentionRangeInDays": {
+          "format": "int32",
+          "description": "Instant RP retention policy range in days",
+          "type": "integer"
+        },
+        "timeZone": {
+          "description": "Time zone optional input as string. For example: \"Pacific Standard Time\".",
+          "type": "string"
+        },
+        "policyType": {
+          "description": "The policy type.",
+          "enum": [
+            "Invalid",
+            "V1",
+            "V2"
+          ],
+          "x-enum-descriptions": [
+            "Invalid Azure IaaS VM Protection Policy type",
+            "V1 Azure IaaS VM Protection Policy type",
+            "V2 Azure IaaS VM Protection Policy type"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IAASVMPolicyType",
+            "modelAsString": true
+          }
+        }
+      },
+      "x-ms-discriminator-value": "AzureIaasVM"
+    },
+    "ProtectionPolicy": {
+      "description": "Base class for backup policy. Workload-specific backup policies are derived from this class.",
+      "required": [
+        "backupManagementType"
+      ],
+      "type": "object",
+      "properties": {
+        "protectedItemsCount": {
+          "format": "int32",
+          "description": "Number of items associated with this policy.",
+          "type": "integer"
+        },
+        "backupManagementType": {
+          "description": "This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.",
+          "type": "string"
+        },
+        "resourceGuardOperationRequests": {
+          "description": "ResourceGuard Operation Requests",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "discriminator": "backupManagementType"
+    },
+    "InstantRPAdditionalDetails": {
+      "type": "object",
+      "description": "Instant recovery point additional details.",
+      "properties": {
+        "azureBackupRGNamePrefix": {
+          "description": "Azure backup resource group name prefix.",
+          "type": "string"
+        },
+        "azureBackupRGNameSuffix": {
+          "description": "Azure backup resource group name suffix.",
+          "type": "string"
+        }
+      }
+    },
+    "SchedulePolicy": {
+      "description": "Base class for backup schedule.",
+      "required": [
+        "schedulePolicyType"
+      ],
+      "type": "object",
+      "properties": {
+        "schedulePolicyType": {
+          "description": "This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.",
+          "type": "string"
```

Comment:
>"type": "string" [](http://example.com/codeflow?start=10&length=16)

Should this be an enum?

Reply:
We need this property to be a string. Azure Backup service has defined this property as string https://github.com/Azure/azure-rest-api-specs/blob/077f1f8f0c8f7e5ac2ab17f9b80d36c56dc5156c/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2023-04-01/bms.json#L9152. Our service is integrated with Azure Backup service. Currently we have defined the backup policy models in our swagger. But before going GA, we would reference the policy models from Azure Backup swagger and not have ownership from our side.